docs/format_howtos.py

```diff
@@ -0,0 +1,78 @@
+"""
+Read all of the HOWTO .diff files and convert them into .html files
+that are both Python syntax highlighted /and/ diff syntax highlighted.
+
+Then these can be included directly in readthedocs as inline HTML
+files.
+"""
+
+import pygments
+import pygments.formatters
+from pygments.lexers import PythonLexer
+
+import os
+
+def main():
+    for diff_filename in os.listdir(os.path.join('..', 'howtos', 'diffs')):
+        format_howto(os.path.join('..', 'howtos', 'diffs', diff_filename),
+                     os.path.join('_formatted_howtos', diff_filename + '.html'))
+
+def format_howto(input_file, output_file):
+    # Load one of our HOWTO diff files.
+    with open(input_file) as f:
+        diff = f.readlines()
```

Comment:
I guess we load the raw .diff file here directly? Would git produce different diffs due to its merging strategy?
Especially the line numbers will quickly bit rot when other parts of the code are updated without causing conflicts.

Reply:
Yes the line number will rot, but git diffs keep metadata about which commit they were forked from which (barring history rewriting on prerelease which we shouldn't do) allows them to be applied later as well. (This is how Git 3 way merge works)

But we don't show the line numbers directly to anyone.

What is true is that at times when the base examples change, then the diff here will be longer than it should be (imagine we brought in part of a HOWTO into the base example). At that point you would want to re-apply and re-pack all of the HOWTOs to make the cleaner.

We could consider doing that as a hook.